File: 1-1000/242_valid_anagrams/solution.py
s = "anagram"
t = "nagaram"


def solve(s: str, t: str) -> bool:
    if len(s) != len(t):
        return False
    dict1, dict2 = dict(), dict()

    for i in range(len(s)):
        dict1[s[i]] = 1 + dict1.get(s[i], 0)
        dict2[t[i]] = 1 + dict2.get(t[i], 0)
    for key in dict1:
        if dict1[key] != dict2.get(key):
            return False
    return True


print(solve(s, t))

File: 1-1000/242_valid_anagrams/solution.go
package main

import "fmt"

func isAnagram(s string, t string) bool {
	dict := make(map[rune]int)

	for _, char := range s {
		dict[char]++
	}
	for _, char := range t {
		dict[char]--
	}
	for _, val := range dict {
		if val != 0 {
			return false
		}
	}
	return true
}

func main() {
	fmt.Println(isAnagram("anagram", "nagaram"))
}
